crates/red_knot_python_semantic/src/types/display.rs

```diff
@@ -92,38 +92,31 @@ impl std::fmt::Display for DisplayRepresentation<'_> {
                 if elements.is_empty() {
                     f.write_str("()")?;
                 } else {
-                    let mut first = true;
-                    for element in &**elements {
-                        if !first {
-                            f.write_str(", ")?;
-                        }
-                        first = false;
-                        element.display(self.db).fmt(f)?;
-                    }
+                    elements.display(self.db).fmt(f)?;
                 }
                 f.write_str("]")
             }
         }
     }
 }
 
-impl<'db> UnionType<'db> {
-    fn display(&'db self, db: &'db dyn Db) -> DisplayUnionType<'db> {
+impl<'a, 'db> UnionType<'db> {
```

Comment:
Is it necessary that we use two lifetimes here considering that all our types in fact do have a `'db` lifetime. 

My preference is only to use as many lifetimes as necessary. Sometimes, it's necessary to use two to make the borrow checker happy (because it is important that some values have different lifetimes), but often, it isn't, and we can use a single lifetime.

Reply:
I think I ran into a different case where we were using `'db` lifetime for `self` where it was wrong and caused a problem, and then got over-aggressive about splitting up lifetimes 😆 I'll change this one back. I feel a bit mixed about it, because in principle the split feels more correct, but you're right that all of our type objects are always db lifetime anyway, so we really don't need to split here.